src/util.jl

```diff
@@ -228,8 +226,8 @@ function interp(x::SVector{D,T}, arr::AbstractArray{T,D}) where {D,T}
     end
     return s
 end
-function interp(x::SVector{D,T}, varr::AbstractArray{T}) where {D,T}
+function interp(x::SVector{D}, varr::AbstractArray) where {D}
     # Shift to align with each staggered grid component and interpolate
-    @inline shift(i) = SVector{D,T}(ifelse(i==j,0.5,0.) for j in 1:D)
-    return SVector{D,T}(interp(x+shift(i),@view(varr[..,i])) for i in 1:D)
+    @inline shift(i) = SVector{D}(ifelse(i==j,0.5f0,0f0) for j in 1:D)
```

Comment:
Does this need to be Float32?

Reply:
It doesn't seem to need to be, no.

Reply:
Read your comment the other way around and didn't fix it in the end!

Reply:
Fixed in https://github.com/weymouth/WaterLily.jl/pull/131.